addons/l10n_cl/models/account_move.py

```diff
@@ -27,23 +26,82 @@ def _get_l10n_latam_documents_domain(self):
+                        latam_document_type_code not in ['110', '111', '112']:
+                    raise ValidationError(_(
+                        'Document types for foreign customers must be export type (codes 110, 111 or 112)'))
+            if rec.journal_id.type == 'sale' and not rec.journal_id.l10n_latam_use_documents:
+                raise ValidationError(_('In Chilean localization, you need to use documents in the sales journal'
+                                        'in order to register your sales properly'))
+            if rec.journal_id.type == 'purchase' and rec.journal_id.l10n_latam_use_documents:
+                if vat != '60805000-0' and latam_document_type_code == '914':
+                    raise ValidationError(_('The DIN document is intended to be used only with RUT 60805000-0'
+                                            ' (Tesorería General de La República)'))
+                if not tax_payer_type or not vat:
+                    if country_id == self.env.ref('base.cl') and latam_document_type_code not in [
+                            '35', '38', '39', '41']:
+                        raise ValidationError(_('Tax payer type and vat number are mandatory for this type of '
+                                                'document. Please set the current tax payer type of this supplier'))
+                if tax_payer_type == '2' and latam_document_type_code not in ['70', '71', '56', '61']:
+                    raise ValidationError(_('The tax payer type of this supplier is incorrect for the selected type'
+                                            ' of document.'))
+                if tax_payer_type in ['1', '3']:
+                    if latam_document_type_code in ['70', '71']:
+                        raise ValidationError(_('The tax payer type of this supplier is not entitled to deliver '
+                                                'fees documents'))
+                    if latam_document_type_code in ['110', '111', '112']:
+                        raise ValidationError(_('The tax payer type of this supplier is not entitled to deliver '
+                                                'imports documents'))
+                if tax_payer_type == '4' or country_id != self.env.ref('base.cl'):
+                    raise ValidationError(_('You need a journal without the use of documents for foreign '
+                                            'suppliers'))
+            if rec.journal_id.type == 'purchase' and not rec.journal_id.l10n_latam_use_documents:
+                if tax_payer_type != '4':
+                    raise ValidationError(_('This supplier should be defined as foreigner tax payer type and '
+                                            'the country should be different from Chile to register purchases.'))
+
+    @api.onchange('journal_id')
+    def _onchange_journal(self):
+        self.l10n_latam_document_type_id = False
+        self._get_l10n_latam_documents_domain()
```

Comment:
Calling this method on itself, does nothing

Reply:
Should not this be done in l10n_latam_invoice_document then?

Reply:
Perhaps should be at latam. But it does something, and think this is important, because if you change the partner / journals it acts and recalculate the domain, and does not present erroneous options.

Reply:
@Danisan Putting the document type to False might have an effect, but the _get_latam_documents_domain does not make sense as the only thing it does, is getting the domain and we don't do anything with it. So I want to remove 103 at least.

Reply:
0f5265b990043254dc0573a2edd3146b6fb00c92 removed here